test/integration/controller_test.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"k8s.io/client-go/1.5/kubernetes/fake"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1alpha1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/brokerapi"
+	fakebrokerapi "github.com/kubernetes-incubator/service-catalog/pkg/brokerapi/fake"
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+	scinformers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers_generated"
+	informers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers_generated/servicecatalog/v1alpha1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/controller/wip"
+	"github.com/kubernetes-incubator/service-catalog/pkg/registry/servicecatalog/server"
+)
+
+// TestController is a very basic test to start
+//
+// need etcd running
+// start a fresh apiserver for the controller to talk to
+func TestController(t *testing.T) {
```

Comment:
Could there be a better name for this? This is testing one or two very specific bits of controller functionality. I assume other test cases that will be forthcoming will occupy their own functions as well... so a more specific naming scheme might be good.